### PR TITLE
c/snap-bootstrap: split CVM related functionality in separate files

### PR DESCRIPTION
**Pull request by sespiros**
As requested in https://github.com/canonical/snapd/pull/14713#pullrequestreview-2473080901

### CODE REVIEW

File: cmd/snap-bootstrap/cmd_initramfs_mounts_cvm_test.go [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+	"github.com/snapcore/snapd/osutil/disks"
+	"github.com/snapcore/snapd/secboot"
+	"github.com/snapcore/snapd/testutil"
+)
+
+var (
+	defaultCVMDisk = &disks.MockDiskMapping{
+		Structure: []disks.Partition{
+			seedPart,
+			cvmEncPart,
```
Comment (pedronis) [inline cmd/snap-bootstrap/cmd_initramfs_mounts_cvm_test.go:39]: should we move cvmEncPart to this file too?

File: cmd/snap-bootstrap/cmd_initramfs_mounts_cvm.go [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+
+func (*genericCVMModel) Classic() bool {
+	return true
+}
+
+func (*genericCVMModel) Grade() asserts.ModelGrade {
+	return "signed"
+}
+
+func generateMountsModeRunCVM(mst *initramfsMountsState) error {
```
Comment (pedronis) [inline cmd/snap-bootstrap/cmd_initramfs_mounts_cvm.go:47]: probably a good idea to have a doc comment attached to this one

**Review by pedronis — COMMENTED** [review]
thank you, couple of small comments

File: cmd/snap-bootstrap/cmd_initramfs_mounts_cvm_test.go (hunk 1 above)
Comment (sespiros) [reply cmd/snap-bootstrap/cmd_initramfs_mounts_cvm_test.go:45]: Thanks, I missed that one. Added.

File: cmd/snap-bootstrap/cmd_initramfs_mounts_cvm.go (hunk 2 above)
Comment (sespiros) [reply cmd/snap-bootstrap/cmd_initramfs_mounts_cvm.go:50]: Added.

**Review by alfonsosanchezbeato — APPROVED** [review]
LGTM, thanks

**Review by pedronis — APPROVED** [review]
thank you

**Pull request merged**